app/src/main/java/eu/kanade/tachiyomi/ui/reader/ReaderActivity.kt

```diff
@@ -1379,13 +1380,16 @@ class ReaderActivity : BaseRxActivity<ReaderPresenter>() {
 
     override fun onProvideAssistContent(outContent: AssistContent) {
         super.onProvideAssistContent(outContent)
-        val manga = presenter.manga ?: return
         val source = presenter.source as? HttpSource ?: return
-        val url = try {
+        val chapterUrl = presenter.getCurrentChapter()?.chapter?.url?.getUrlWithoutDomain()
+        val url = if (chapterUrl.isNullOrBlank()) try {
+            val manga = presenter.manga ?: return
             source.mangaDetailsRequest(manga).url.toString()
         } catch (e: Exception) {
+            toast(e.message)
             return
-        }
+        } else source.baseUrl + chapterUrl
```

Comment:
i'd just create a helper func in presenter for building this chapter url returning a `String?`, returning if its null

No need for the toast catch either since it should never enter the catch case, the try is more of a safety measure

Reply:
Done
Not sure if it was exactly what you meant but I'll let you judge ([5e013c0](https://github.com/Jays2Kings/tachiyomiJ2K/pull/1175/commits/5e013c0acb02633e691ef9d39d33a387f754a4fe))